lib/ConsumerImpl.cc

```diff
@@ -460,6 +476,7 @@ void ConsumerImpl::messageReceived(const ClientConnectionPtr& cnx, const proto::
         return;
     }
 
+    auto redeliveryCount = msg.redelivery_count();
```

Comment:
It seems that we need to pass the `redeliveryCount` to the `decryptMessageIfNeeded` method before? When the decryption failed, `unAckedMessageTracker.add(m, redeliveryCount);` should be called. (Though I didn't find if there is a related test)

Reply:
It seems that this is another problem: we need to give the client that cannot decrypt the message a chance to re-decrypt the message, so we need to put it in `unAckedMessageTracker ` and wait for re-delivery.

https://github.com/apache/pulsar/pull/3097 


As for why `redeliveryCount` is passed into `decryptPayloadIfNeeded` methods in the Java implementation, it is because of the support for [PIP-130](https://github.com/apache/pulsar/pull/13707)


Note: I'll open a new PR to support both issues.